### ISSUE TITLE
Get NSURLErrorCancelled error when try to download one image, but use AFNetworking works

### ISSUE THREAD

**Issue by HaleyYQ** [rs/SDWebImage#1922]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

I use sd_setImageWithURL: to download the image with the URL https://sandbox.dev.clover.com/v2/image/Red_209506569580589592.jpg, it returned the following error. 
But use AFNetworking API setImageWithURL: works.
It seems that SDWebImage doesn't support to download the URL which is needed to redirect the URL.

If there is one solution for this issue? or improve SDWebImage to support this case.

Error Domain=NSURLErrorDomain Code=-999 "cancelled" UserInfo={NSErrorFailingURLKey=https://sandbox.dev.clover.com/v2/image/Red_209506569580589592.jpg, NSLocalizedDescription=cancelled, NSErrorFailingURLStringKey=https://sandbox.dev.clover.com/v2/image/Red_209506569580589592.jpg}

**Comment by mythodeia** [comment]
@HaleyYQ did you try the url in the demo project? which version are you using? download the latest version and try again

**Comment by HaleyYQ** [comment]
@mythodeia I tried to user the SDWebImage version(4.0.0 and 3.8.2), it still doesn't work. This URL seems that it is different from others. When click the URL, Web browse will pop up to download it in web. But other URLs is displayed in the Web browse.

**Comment by mythodeia** [comment]
@HaleyYQ this screenshot below is from the latest version. I tried the demo project with your url and it shows just fine. Thats why i asked you to test the url in the demo project.
<img width="375" alt="screenshot" src="https://cloud.githubusercontent.com/assets/1748458/26712399/5a90938c-476f-11e7-90a7-044adfa0b4ff.png">

**Comment by HaleyYQ** [comment]
@mythodeia Thanks your help. The demo works for the link. However, I copied the SDWebImage framework from demo to my project, and use FLAnimatedImageView, it still returned NSURLErrorCancelled. Do I miss something?

**Comment by mythodeia** [comment]
@HaleyYQ the demo project uses FLAnimatedImage too. maybe there is something else that causes the issue in your project.

**Comment by HaleyYQ** [comment]
@mythodeia  I got the point of why it got different result. In the demo, it used the code [SDWebImageManager sharedManager].imageDownloader.username = @"httpwatch"; [SDWebImageManager sharedManager].imageDownloader.password = @"httpwatch01"; If you remove the two lines code, the demo also returned the NSURLErrorCancelled error. And I add the two lines code to my project, it works.
One question, what's effect is the two lines code? It is magical。

**Comment by mythodeia** [comment]
thats the HTTP NTLM auth example. it takes a username/password where you can provide in those two lines you mentioned
